compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala

```diff
@@ -570,13 +570,14 @@ class ClassfileParser(
         case tpnme.AnnotationDefaultATTR =>
           sym.addAnnotation(Annotation(defn.AnnotationDefaultAnnot, Nil))
         // Java annotations on classes / methods / fields with RetentionPolicy.RUNTIME
-        case tpnme.RuntimeAnnotationATTR =>
+        case tpnme.RuntimeVisibleAnnotationATTR
+          | tpnme.RuntimeInvisibleAnnotationATTR =>
           parseAnnotations(attrLen)
 
         // TODO 1: parse runtime visible annotations on parameters
         // case tpnme.RuntimeParamAnnotationATTR
 
-        // TODO 2: also parse RuntimeInvisibleAnnotation / RuntimeInvisibleParamAnnotation,
```

Comment:
This TODO also exists in Scala 2 (https://github.com/scala/scala/blob/ec46ff84333501bd11c8cac89802508d580c470d/src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala#L859). @retronym : any idea why these annotations have never been parsed so far ?

Reply:
No good reason that I'm aware of. I agree we should parse these and give then `StaticAnnotation` parent.